### DISCUSSION TITLE
How does starship hook into terminal resize events?

### DISCUSSION THREAD

**Discussion by deg0nz** [starship/starship#3978] [Q&A] [answered]
Hi,

I observed, that starship updates the battery state when the terminal is resized.

This kind of behaviour is needed for an automatic re-rendering feature in [mitsuhiko/dialoguer](https://github.com/mitsuhiko/dialoguer), but I wasn't able to find a good solution for this yet.

Can anyone point me to the part in the starship code that implements the "resize-watcher"? 
I searched the repo, but I wasn't able to find out how starship does this.

Thanks and best regards
Beh

**Comment by chipbuster** [comment, marked as answer]
Hi @deg0nz,

We don't actually do this ourselves. If you try this in bash or elvish, you should find that the starship prompt doesn't update on window change (it's much more noticeable if you use the time module with second-level timestamps).

I think this is accomplished in zsh by a SIGWINCH handler, but I don't know if there's general approach to handling this across all shells. My gut says the sophistication of shell/terminal interactions just isn't there yet.

**Reply by deg0nz** [reply]
Ok, thanks for the info!

I already had an idea to implement this via the `SIGWINCH` signal for *nix, but I had the idea that maybe starship does something neat to observe the signal. So I will follow this path further...
